Core/clim-basic/drawing/pattern.lisp

```diff
@@ -99,42 +99,50 @@ pattern, stencil, image etc)."))
   ((array :type (simple-array (unsigned-byte 32) 2)))
   (:documentation "Helper class of RGBA result of another pattern."))
 
+(declaim (ftype (function (t) (values (unsigned-byte 32) &optional nil))
+                %rgba-value))
 (defun %rgba-value (element)
   "Helper function collapsing uniform design into 4-byte RGBA value."
   (flet ((transform (parameter)
-           (logand (logand (truncate (* parameter 255)) 255))))
+           (logand (truncate (* parameter 255)) 255)))
     (etypecase element
       ((unsigned-byte 32) element)
       ;; Uniform-compositium is a masked-compositum rgb + opacity
       ((or color opacity uniform-compositum)
        (multiple-value-bind (red green blue opacity)
            (color-rgba element)
-         (logior (ash (transform red)   24)
-                 (ash (transform green) 16)
-                 (ash (transform blue)   8)
-                 (transform opacity))))
+         (logior (ash (transform red)    24)
+                 (ash (transform green)  16)
+                 (ash (transform blue)    8)
+                 (ash (transform opacity) 0))))
```

Comment:
what's the point of calling ash here?

Reply:
Readability. The call surely gets optimized away.